fe/src/main/java/org/apache/doris/http/rest/RestCountAction.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.doris.catalog.Partition;
+import org.apache.doris.catalog.Replica;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.catalog.Tablet;
+import org.apache.doris.cluster.ClusterNamespace;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.DorisHttpException;
+import org.apache.doris.http.ActionController;
+import org.apache.doris.http.BaseRequest;
+import org.apache.doris.http.BaseResponse;
+import org.apache.doris.http.IllegalArgException;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is responsible for fetch the approximate row count of the specified table from cluster-meta data,
+ * the approximate row maybe used for some computing system to decide use which compute-algorithm can be used
+ * such as shuffle join or broadcast join.
+ * <p>
+ * This API is not intended to compute the exact row count of the specified table, if you need the exact row count,
+ * please consider using the sql syntax `select count(*) from {table}`
+ */
+public class RestCountAction extends RestBaseAction {
+    public RestCountAction(ActionController controller) {
+        super(controller);
+    }
+
+    public static void registerAction(ActionController controller) throws IllegalArgException {
+        controller.registerHandler(HttpMethod.GET,
+                "/api/{cluster}/{database}/{table}/_count", new RestCountAction(controller));
+        controller.registerHandler(HttpMethod.GET,
+                "/api/{database}/{table}/_count", new RestCountAction(controller));
+    }
+
+    @Override
+    protected void executeWithoutPassword(ActionAuthorizationInfo authInfo, BaseRequest request, BaseResponse response) throws DdlException {
+        // just allocate 2 slot for top holder map
+        Map<String, Object> resultMap = new HashMap<>(4);
+        String clusterName = request.getSingleParameter("cluster");
+        String dbName = request.getSingleParameter("database");
+        String tableName = request.getSingleParameter("table");
+        try {
+            if (Strings.isNullOrEmpty(clusterName)
+                    || Strings.isNullOrEmpty(dbName)
+                    || Strings.isNullOrEmpty(tableName)) {
+                throw new DorisHttpException(HttpResponseStatus.BAD_REQUEST, "{cluster}/{database}/{table} must be selected");
+            }
+            String fullDbName = ClusterNamespace.getFullName(authInfo.cluster, dbName);
+            // check privilege for select, otherwise return HTTP 401
+            checkTblAuth(authInfo, fullDbName, tableName, PrivPredicate.SELECT);
+            Database db = Catalog.getCurrentCatalog().getDb(fullDbName);
+            if (db == null) {
+                throw new DorisHttpException(HttpResponseStatus.NOT_FOUND, "Database [" + dbName + "] " + "does not exists");
+            } else {
```

Comment:
I think removing `else` clause is better

Reply:
Yeah